src/main/rx/crsf.c

```diff
@@ -184,46 +185,52 @@ STATIC_UNIT_TESTED void crsfDataReceive(uint16_t c, void *data)
 
     if (crsfFramePosition < fullFrameLength) {
         crsfFrame.bytes[crsfFramePosition++] = (uint8_t)c;
-        crsfFrameDone = crsfFramePosition < fullFrameLength ? false : true;
-        if (crsfFrameDone) {
+        if (crsfFramePosition >= fullFrameLength) {
             crsfFramePosition = 0;
-            if (crsfFrame.frame.type != CRSF_FRAMETYPE_RC_CHANNELS_PACKED) {
-                const uint8_t crc = crsfFrameCRC();
-                if (crc == crsfFrame.bytes[fullFrameLength - 1]) {
-                    switch (crsfFrame.frame.type)
-                    {
+            const uint8_t crc = crsfFrameCRC();
+            if (crc == crsfFrame.bytes[fullFrameLength - 1]) {
+                switch (crsfFrame.frame.type)
+                {
+                    case CRSF_FRAMETYPE_RC_CHANNELS_PACKED:
+                        if (crsfFrame.frame.deviceAddress == CRSF_ADDRESS_FLIGHT_CONTROLLER) {
+                            crsfFrameDone = true;
+                            memcpy(&crsfChannelDataFrame, &crsfFrame, sizeof(crsfFrame));
```

Comment:
Codacy found an issue: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6310371)

Reply:
interesting.